### ISSUE TITLE
Unable to run same multi browser tests with OpenSDK features

### ISSUE THREAD

**Issue by kishangupta-15** [testproject-io/csharp-opensdk#144]
I saw example code from #98. However, if I use it as per example for Multiple browser testing, I won't be able to use OpenSDK driver features, because the driver object IWebDriver which doesn't has the definitions for report and other features of OpenSDK.

I have added below line to your code from #98:
**this.driver.Report().Step("End Tests, "Test Ends Here");**

But as expected, this gives error: "Error CS1061 'IWebDriver' does not contain a definition for 'Report' and no accessible extension method 'Report' accepting a first argument of type 'IWebDriver' could be found (are you missing a using directive or an assembly reference?)"

Could you please help in resolving this?

```csharp
namespace TestProject.OpenSDK.Tests.Examples.Frameworks.NUnit
{
    using global::NUnit.Framework;
    using OpenQA.Selenium;
    using TestProject.OpenSDK.Drivers.Web;
    using TestProject.OpenSDK.Internal.Helpers.DriverOptions;

    [TestFixture(BrowserType.Chrome)]
    [TestFixture(BrowserType.Firefox)]
    public class MultipleBrowserTest
    {
        private IWebDriver driver;
        private BrowserType browserType;

        public MultipleBrowserTest(BrowserType type)
        {
            this.browserType = type;
        }

        [SetUp]
        public void StartBrowser()
        {
            switch (this.browserType)
            {
                case BrowserType.Chrome:
                    this.driver = new ChromeDriver();
                    break;
                case BrowserType.Firefox:
                    this.driver = new FirefoxDriver();
                    break;
                default:
                    throw new System.Exception("Unsupported browser type");
            }
        }

        public void ExampleTestUsingChromeDriver()
        {
            this.driver.Navigate().GoToUrl("https://example.testproject.io");
            this.driver.FindElement(By.CssSelector("#name")).SendKeys("John Smith");
            this.driver.FindElement(By.CssSelector("#password")).SendKeys("12345");
            this.driver.FindElement(By.CssSelector("#login")).Click();
            this.driver.Report().Step("End Tests, "Test Ends Here"); //driver type is IWebDriver which doesn't have OpenSDK functions

            Assert.IsTrue(this.driver.FindElement(By.CssSelector("#greetings")).Displayed);
        }
    }
}
```

**Comment by basdijkstra** [comment]
Thank you for letting me know. I'll see what I can do here, this might not be too difficult to add.

**Comment by basdijkstra** [comment]
I just submitted #145, which implements this enhancement. To use it once this PR is merged, use the `IWebDriver` interface from the `TestProject.OpenSDK.Drivers` namespace and you should be good to go. Here's a test result from your test on my machine after implementing the enhancement:

![image](https://user-images.githubusercontent.com/10740451/111795214-44daee00-88c7-11eb-8f0c-931a1e560002.png)

One test uses Chrome, the other Firefox (this isn't reflected in the test name, though, because that's inferred from the method name).

**Comment by kishangupta-15** [comment]
@basdijkstra I see that it works for Web Drivers. What about Generic Driver, should it also not implement the same interface as the other drivers so that all the driver related initialization can happen in the same class such as DriverFactory?

**Comment by basdijkstra** [comment]
> @basdijkstra I see that it works for Web Drivers. What about Generic Driver, should it also not implement the same interface as the other drivers so that all the driver related initialization can happen in the same class such as DriverFactory?

That's not something I'd like to do, because that would mean that the generic driver would also have to extend the Selenium IWebDriver interface and just throw NotImplemented errors for all the Selenium methods (FindElement etc.).

This would give the suggestion that the generic driver is a driver that can be used for web / mobile applications, which it is not, and would only lead to confusion for users. 

I'm open to suggestions but for now I'm keeping the generic driver out of this.

**Comment by basdijkstra** [comment]
@kishangupta-15 this fix is included in 0.66.0, so please retest with that version and close this issue when you're happy with the fix and my comment above this one. Thanks.

**Comment by kishangupta-15** [comment]
@basdijkstra I have retested it with v 0.66.0 and it works fine. Thanks for the quick fix. Closing the issue.
